### PR TITLE
Simple file upload download

### PR DESCRIPTION
**Pull request by matt-sm**
- didn't need the complexity over in S3.py

### CODE REVIEW

File: t [hunk 1]
```diff
@@ -0,0 +1,52 @@
+boto==2.42.0
```
Comment (maxious) [inline t]: is this file needed?

File: dmutils/file.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import os
+import boto3
+from werkzeug.utils import secure_filename
+from flask import current_app
+from io import BytesIO
+
+
+def allowed_file(filename):
+    return '.' in filename and \
```
Comment (djrobstep) [inline dmutils/file.py]: Doesn't secure_filename do file path checking already?

File: dmutils/file.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        filename.rsplit('.', 1)[1] in current_app.config.get('ALLOWED_EXTENSIONS')
+
```
Comment (djrobstep) [inline dmutils/file.py:10]: This checks the filename only which doesn't necessarily match the file contents.. Worth checking mime type of file?

File: tests/test_file.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import pytest
+import mock
+from dmutils.config import init_app
+from dmutils.file import s3_upload_fileObj, s3_upload_file_from_request, s3_download_file
+
+
+@pytest.yield_fixture
```
Comment (djrobstep) [inline tests/test_file.py]: yield_fixture is deprecated in the latest pytest version: http://blog.pytest.org/2016/whats-new-in-pytest-30/

File: tests/test_file.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+def s3_resource():
+    with mock.patch('boto3.resource') as boto_resource:
+        instance = boto_resource.return_value
+        yield instance
+
+
+@pytest.yield_fixture
+def file_app(app):
+    init_app(app)
+    app.config['ALLOWED_EXTENSIONS'] = ['pdf']
```
Comment (djrobstep) [inline tests/test_file.py:17]: Should the extension include the dot?

**Review by djrobstep — APPROVED** [review]
